EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -63,11 +78,19 @@ private enum GuildDialogState
             Title = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_MASTER);
         }
 
-        private void SetState(GuildDialogState state)
+        protected override void OnUpdateControl(GameTime gameTime)
         {
-            if (state != GuildDialogState.Initial && _state == state)
-                return;
+            if (_guildDescription != _guildSessionProvider.GuildDescription && _guildSessionProvider.GuildDescription != "")
+            {
+                _guildDescription = _guildSessionProvider.GuildDescription;
+                SetState(GuildDialogState.Modify);
```

Comment:
`SetState` as part of the update doesn't really make sense to me. IMO it should be the result of an action the user takes. Is guild dialog supposed to enter the `Modify` state when the description is changed?

Reply:
This was more of a refresh the get the description to properly show

Reply:
Now tracking the list item in a private optional field as suggested